e2e/_suites/fleet/services.go

```diff
@@ -166,84 +161,17 @@ func runElasticAgentCommand(profile string, image string, service string, proces
 // to be used will be defined by the local snapshot produced by the local build.
 // Else, if the environment variable BEATS_USE_CI_SNAPSHOTS is set, then the artifact
 // to be downloaded will be defined by the latest snapshot produced by the Beats CI.
-func downloadAgentBinary(artifact string, version string, OS string, arch string, extension string, stale bool) (string, string, error) {
-	fileName := fmt.Sprintf("%s-%s-%s.%s", artifact, version, arch, extension)
-
-	beatsLocalPath := shell.GetEnv("BEATS_LOCAL_PATH", "")
-	if beatsLocalPath != "" {
-		distributions := path.Join(beatsLocalPath, "x-pack", "elastic-agent", "build", "distributions")
-		log.Debugf("Using local snapshots for the Elastic Agent: %s", distributions)
-
-		if extension == "tar.gz" {
-			fileName = fmt.Sprintf("%s-%s-%s-%s.%s", artifact, version, OS, arch, extension)
-		}
-
-		fileNamePath := path.Join(distributions, fileName)
-		_, err := os.Stat(fileNamePath)
-		if err != nil || os.IsNotExist(err) {
-			return fileName, fileNamePath, err
-		}
-
-		return fileName, fileNamePath, err
-	}
-
-	handleDownload := func(URL string, fileName string) (string, string, error) {
-		if val, ok := binariesCache[URL]; ok {
-			log.WithFields(log.Fields{
-				"URL":  URL,
-				"path": val,
-			}).Debug("Retrieving binary from local cache")
-			return fileName, val, nil
-		}
-
-		filePath, err := e2e.DownloadFile(URL)
-		if err != nil {
-			return fileName, filePath, err
-		}
-
-		binariesCache[URL] = filePath
-
-		return fileName, filePath, nil
-	}
-
-	if downloadURL, exists := os.LookupEnv("ELASTIC_AGENT_DOWNLOAD_URL"); exists {
-		return handleDownload(downloadURL, fileName)
-	}
-
-	var downloadURL string
-	var err error
-
-	useCISnapshots := shell.GetEnvBool("BEATS_USE_CI_SNAPSHOTS")
-	if useCISnapshots {
-		log.Debug("Using CI snapshots for the Elastic Agent")
-
-		bucketFileName, bucket, prefix, object := getGCPBucketCoordinates(fileName, artifact, version, OS, arch, extension, stale)
-
-		maxTimeout := time.Duration(timeoutFactor) * time.Minute
-
-		downloadURL, err = e2e.GetObjectURLFromBucket(bucket, prefix, object, maxTimeout)
-		if err != nil {
-			return "", "", err
-		}
-
-		return handleDownload(downloadURL, bucketFileName)
-	}
-
-	downloadVersion := version
-	if !stale {
-		downloadVersion = checkElasticAgentVersion(version)
-	}
-
-	downloadURL, err = e2e.GetElasticArtifactURL(artifact, downloadVersion, OS, arch, extension)
+func downloadAgentBinary(artifactName string, artifact string, version string) (string, error) {
```

Comment:
I could have remove this method, but instead I decide to internally call the new method